### ISSUE TITLE
winetricks -q windowscodecs  failes on  WINEARCH=win32 (Regression)

### ISSUE THREAD

**Issue by bitranox** [Winetricks/winetricks#1260]
it wortked before, but now stopps working : 

Using winetricks 20190615-next - sha256sum: 91c449ac44acecc23b2c0917c4e0edaf2c1d71f2e4356f3b1810e64474a304e5 with wine-4.0.1 and WINEARCH=win32
Command "winetricks -q windowscodecs" failed.

Using winetricks 20190615-next - sha256sum: 91c449ac44acecc23b2c0917c4e0edaf2c1d71f2e4356f3b1810e64474a304e5 with wine-4.10 and WINEARCH=win32
Command "winetricks -q windowscodecs" failed.

Using winetricks 20190615-next - sha256sum: 91c449ac44acecc23b2c0917c4e0edaf2c1d71f2e4356f3b1810e64474a304e5 with wine-4.10 (Staging) and WINEARCH=win32
Command "winetricks -q windowscodecs" failed.

on 64 Bit architecture it still works !

You might look at my travis build : 

https://travis-ci.org/bitranox/lib_platform

here the last working : 

Using winetricks 20190615-next - sha256sum: 9e8cec8015975960fa3358b6e374d54bf3ef1037ff981dcb2d3f52487c387b2d with wine-4.0.1 and WINEARCH=win32

Using winetricks 20190615-next - sha256sum: 9e8cec8015975960fa3358b6e374d54bf3ef1037ff981dcb2d3f52487c387b2d with wine-4.10 and WINEARCH=win32

Using winetricks 20190615-next - sha256sum: 9e8cec8015975960fa3358b6e374d54bf3ef1037ff981dcb2d3f52487c387b2d with wine-4.10 (Staging) and WINEARCH=win32

You might check at : 
https://travis-ci.org/bitranox/install_python_on_wine_and_travis

**Comment by rdimesio** [comment]
I can confirm this, and it's not just this verb; current winetricks fails in general in a 32 bit wineprefix. Reported on the WineHQ forum: https://forum.winehq.org/viewtopic.php?f=2&t=32544

**Comment by austin987** [comment]
Thanks for the report; I was worried this was in the latest release, but that's fine, it was the last commit I did (which I had tested earlier commits on 32-bit, but the last wasn't tested closely enough).

Anyway, please try https://github.com/austin987/winetricks/pull/new/fix64-metadata

it fixes the issue for me, but I'd like to look at things a bit closer before committing.

**Comment by bitranox** [comment]
Why You dont set up something like  https://travis-ci.org/bitranox/install_python_on_wine_and_travis
to automatically test on travis all that stuff ?

**Comment by austin987** [comment]
I tried at one point, which is why winetricks-tests has the huge XVFB_BLACKLIST and ARHCIVE_ORG_BLACKLIST:
https://github.com/Winetricks/winetricks/blob/master/tests/winetricks-test#L106

most things didn't work under xvfb (at the time at least). I see that windowscodecs works for you, it failed for me. Retrying, seems it gets a non-fatal error:
```
XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":99"
      after 186 requests (183 known processed) with 0 events remaining.
austin@laptop:~/src/winetricks$ echo $?
0
```

however a full run of the tests takes ~3 hours for me on a real pc, which will definitely exceed the time for travis-ci.org:
https://docs.travis-ci.com/user/customizing-the-build

In short, didn't seem very feasible, it may be possible to do some very basic tests, but it's not clear what those should be.

**Comment by austin987** [comment]
With just the current xvfb tests, on my laptop with everything already downloaded/cached:

```
===============================================================
Test over, 0 failures, 192 successes, 6 skipped.
PASS
===============================================================

real	76m9.168s
user	11m35.871s
sys	5m45.080s
```

**Comment by bitranox** [comment]
well, that would be definitely too long for me ;-) 
I think multiprocessing to utilize all cores, and configure on 8 wine instances (or more) in parallel should bring the time down considerably. As nowadays python3 is as ubiquitous as bash, consider to rewrite the test in python - it will make the life much easier I guess.
Since Your original script is only a thousand lines, I might give it a shot.
What is the CPU and HDD Utilisation during the test ?
(I assume You have at least 4 cores and SSD ?)

**Comment by austin987** [comment]
This is fixed by https://github.com/Winetricks/winetricks/commit/0fd91cbc999faf6f269109df2c71d5027d0a664c

Discussions about travis should really be in a separate bug. That said, I didn't track utilization, but you should keep in mind travis only has two cores. In addition, if making the tests parallel, there's a race condition to consider for verbs that share cached files (e.g., directx or anything from service packs).

**Comment by bitranox** [comment]
First of all, thank You for fixing !

>>> Discussions about travis should really be in a separate bug. 
Agree.
>>> but you should keep in mind travis only has two cores. In addition, if making the tests parallel, there's a race condition to consider for verbs that share cached files

my approach would be : 
#1 create wine environment as a template
#2 copy that template to a number of new, fresh environments (one for each parallel running test)
#3 run one test in each environment indipendently
#4 after finishing one test, delete the environment and copy a new one from the template
#5 run the next test .... 

even if it is only two cores, there should be some speedup - I will give it a try when i find some time and let You know. And not all operations are CPU bound.
It might pay off for You, if You run the test like that on a local, more powerful machine - my desktop has 12 cores ;-)
